daisy/workflow/validate.go

```diff
@@ -141,6 +142,9 @@ func (w *Workflow) validateRequiredFields() error {
 	if w.Name == "" {
 		return errors.New("must provide workflow field 'Name'")
 	}
+	if !rfc1035Rgx.MatchString(strings.ToLower(w.Name)) {
```

Comment:
I've wondered about a Populate() method, but it might not be necessary in this case. We would just have to change genName() to sanitize the workflow name.